pkg/statistics/handle/ddl/ddl.go

```diff
@@ -195,6 +199,173 @@ func (h *ddlHandlerImpl) HandleDDLEvent(t *util.DDLEvent) error {
 	return nil
 }
 
+func (h *ddlHandlerImpl) onExchangeAPartition(t *util.DDLEvent) error {
+	globalTableInfo, originalPartInfo, originalTableInfo := t.GetExchangePartitionInfo()
+	// Get the count of the partition.
+	partCount, partModifyCount, err := h.statsWriter.StatsMetaCountAndModifyCount(
+		// You only can exchange one partition at a time.
+		originalPartInfo.Definitions[0].ID,
+	)
+	if err != nil {
+		return err
+	}
+	// Get the count of the table.
+	tableCount, tableModifyCount, err := h.statsWriter.StatsMetaCountAndModifyCount(
+		originalTableInfo.ID,
+	)
+	if err != nil {
+		return err
+	}
+
+	// The count of the partition should be added to the table.
+	// The formula is: total_count = original_table_count - original_partition_count + new_table_count.
+	// So the delta is : new_table_count - original_partition_count.
+	countDelta := tableCount - partCount
+	// Initially, the sum of tableCount and partCount represents
+	// the operation of deleting the partition and adding the table.
+	// Therefore, they are considered as modifyCountDelta.
+	// Next, since the old partition no longer belongs to the table,
+	// the modify count of the partition should be subtracted.
+	// The modify count of the table should be added as we are adding the table as a partition.
+	modifyCountDelta := (tableCount + partCount) - (partModifyCount + tableModifyCount)
+
+	// Update the global stats.
+	if modifyCountDelta != 0 || countDelta != 0 {
+		se, err := h.statsHandler.SPool().Get()
+		if err != nil {
+			return errors.Trace(err)
+		}
+		sctx := se.(sessionctx.Context)
+		is := sctx.GetDomainInfoSchema().(infoschema.InfoSchema)
+		schema, _ := is.SchemaByTable(globalTableInfo)
```

Comment:
Though this should not happen, I think it's better to handle the failed to get `DBInfo` case.

Reply:
To make it a bit more rigorous 😆 would it be possible for the non-partitioned table and the partitioned table to come from different databases?

Reply:
Nice point. It is possible. I will add it.

Reply:
It's difficult to retrieve the original schema since we've already exchanged the metadata. Therefore, we won't be able to use any API to locate it. As a result, I'll skip this step.

Reply:
Got it.

Reply:
Looks like we'll need either "old infoschema + old normal table info" or "new inforschema + new normal table info" to get it. I think it's doable but I'm OK to leave it to the future.